plugin-annotation/src/main/java/com/mapbox/maps/plugin/annotation/AnnotationManagerImpl.kt

```diff
@@ -162,7 +165,7 @@ abstract class AnnotationManagerImpl<G : Geometry, T : Annotation<G>, S : Annota
     }
   }
 
-  protected fun initLayerAndSource() {
+  @Synchronized protected fun initLayerAndSource() {
```

Comment:
Synchronized not needed, all those events will happen on main thread.
https://github.com/mapbox/mapbox-maps-android/blob/da2ea096429959c2159a955042d5bfc222763922/sdk/src/main/java/com/mapbox/maps/NativeMapObserver.kt#L31-L33

Reply:
In the last commit, we got the following error on firebase, it seems we have sync issue in MapChange and getStyle callback. Maybe we could remove one to remove this `@Synchronized`, need to rethink it.

`java.lang.RuntimeException: Add source failed: Source mapbox-android-circle-source-1 already exists
	at com.mapbox.maps.extension.style.sources.Source.bindTo(Source.kt:59)
	at com.mapbox.maps.extension.style.sources.SourceKt.addSource(Source.kt:185)
	at com.mapbox.maps.plugin.annotation.AnnotationManagerImpl.initLayerAndSource(AnnotationManagerImpl.kt:177)
	at com.mapbox.maps.plugin.annotation.AnnotationManagerImpl$1$1.invoke(AnnotationManagerImpl.kt:114)
	at com.mapbox.maps.plugin.annotation.AnnotationManagerImpl$1$1.invoke(AnnotationManagerImpl.kt:42)
	at com.mapbox.maps.plugin.MapDelegateProviderImpl$getStyle$1.onStyleLoaded(MapDelegateProviderImpl.kt:20)
	at com.mapbox.maps.MapboxMap.getStyle(MapboxMap.kt:226)`

Reply:
Would be great to reproduce the error and find the root cause but for me it seems that `synchronized` is not needed here (if looking at code) and may just find the root cause for that error even harder to detect in future.

Reply:
Yeah, find a new solution to remove `synchronized`, and works well.